src/test/java/com/tabnine/integration/ConnectionLostNotificationTests.java

```diff
@@ -0,0 +1,86 @@
+package com.tabnine.integration;
+
+import static com.tabnine.testUtils.StateChangedTopicUtils.getMockedStateResponse;
+import static com.tabnine.testUtils.StateChangedTopicUtils.notifyStateChangedTopic;
+
+import com.intellij.notification.Notification;
+import com.intellij.notification.Notifications;
+import com.intellij.openapi.application.ApplicationManager;
+import com.tabnine.MockedBinaryCompletionTestCase;
+import com.tabnine.binary.requests.config.CloudConnectionHealthStatus;
+import com.tabnine.notifications.ConnectionLostNotificationHandler;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.jetbrains.annotations.NotNull;
+import org.junit.Test;
+
+public class ConnectionLostNotificationTests extends MockedBinaryCompletionTestCase {
+
+  @Test
+  public void should_show_connection_lost_notification_when_connection_unhealthy() {
+    ApplicationManager.getApplication()
+        .getMessageBus()
+        .connect()
+        .subscribe(
+            Notifications.TOPIC,
+            new Notifications() {
+              @Override
+              public void notify(@NotNull Notification notification) {
+                assertTrue(notification.getContent().contains("Tabnine lost internet connection"));
```

Comment:
extract const, or even better - use the actual message const if possible

Reply:
I'm not sure if this is a good practice to mark a variable as public only for test purposes. wdyt?